sympy/tensor/array/ndim_array.py

```diff
@@ -251,10 +253,15 @@ def _eval_derivative(self, arg):
     def _eval_derivative_array(self, arg):
         from sympy import derive_by_array
         from sympy import Tuple
+        from sympy import SparseNDimArray
         from sympy.matrices.common import MatrixCommon
+
         if isinstance(arg, (Iterable, Tuple, MatrixCommon, NDimArray)):
             return derive_by_array(self, arg)
         else:
+            if isinstance(self, SparseNDimArray):
```

Comment:
This method was not supposed to be modified, applyfunc should be able to handle sparse arrays efficiently.